include/nbl/system/IAsyncQueueDispatcher.h

```diff
@@ -19,10 +19,20 @@ namespace impl
             // wait on this for result to be ready
             std::condition_variable cvar;
             bool ready = false;
+
+            void lock() {}
+            void unlock() {}
```

Comment:
what's the purpose of this!?

Reply:
for ISkippableAsyncQueueDispatcher, so that i can have lock()/unlock() logic here and not repeat code in Skippable

Reply:
these functions will be overriden anyway, so they'll either have to be called via `static_cast<CRTP::request_t&>(req).begin()` or be polymorphic

anyway probably better to call them `begin()` and `end()`

Reply:
> these functions will be overriden anyway, so they'll either have to be called via static_cast<CRTP::request_t&>(req).begin() or be polymorphic

ofc, they are

Reply:
resolved.